test/index.js

```diff
@@ -1601,6 +1601,68 @@ describe('Schwifty', () => {
                 expect(c.validate('x')).to.equal({ value: 'x' });
                 expect(c.validate(1)).to.contain('error');
             });
+
+            it('supports nested properties', () => {
+
+                const Model = class extends Schwifty.Model {
+                    static get joiSchema() {
+
+                        return Joi.object({
+                            a: Joi.object({
+                                d: Joi.string().min(3),
+                                e: Joi.string().default('e')
+                            }),
+                            b: Joi.string().default('b'),
+                            c: Joi.string().required()
+                        });
+                    }
+                };
+
+                const d = Model.field('a.d');
+                const e = Model.field('a.e');
+
+                const dfull = Model.field('a.d').tailor('full');
+                const efull = Model.field('a.e').tailor('full');
+
+                expect(d.validate('123')).to.equal({ value: '123' });
+                expect(d.validate('12')).to.contain('error');
+
+                expect(e.validate()).to.equal({ value: undefined });
+                expect(e.validate('x')).to.equal({ value: 'x' });
+                expect(e.validate(1)).to.contain('error');
+
+                expect(dfull.validate('123')).to.equal({ value: '123' });
+                expect(dfull.validate('12')).to.contain('error');
+
+                expect(efull.validate()).to.equal({ value: 'e' });
+                expect(efull.validate('x')).to.equal({ value: 'x' });
+                expect(efull.validate(1)).to.contain('error');
+            });
+
+            it('throws when called on a field whose schema contains a ref', () => {
```

Comment:
I wonder if it would be nice to rephrase this to indicate that `field()` doesn't actually throw.  Validation also doesn't _necessarily_ throw, e.g. if the field is plopped into a larger schema where the ref does exist.  This schema could work just fine, for example:

```js
Joi.object({
    a: Model.field('a'),
    c: Model.field('c')
});
```

Reply:
Woah, no way! I didn't realize that was possible, thanks for pointing that technique out.
And for the suggested rewording, that makes sense. I just pushed up some changes that hopefully addressed those points 🙏